### PR TITLE
Master stock on hand sle

### PR DESCRIPTION
**Pull request by sle-odoo**

### CODE REVIEW

File: addons/stock/models/stock_quant.py [hunk 1]
```diff
@@ -173,6 +190,22 @@ def write(self, vals):
...
         return super(StockQuant, self).write(vals)
 
+    def fields_view_get(self, view_id=None, view_type='form', toolbar=False, submenu=False):
+        res = super().fields_view_get(view_id=view_id, view_type=view_type, toolbar=toolbar, submenu=submenu)
+        if res['type'] == 'search' and self.env.context.get('onhand_filter_domain'):
+            doc = etree.XML(res['arch'])
+            magic_filter_attrib = {
+                'name': 'onhand',
+                'string': 'On Hand',
+                'domain': str(self.env.context['onhand_filter_domain']),
```
Comment (sle-odoo) [inline addons/stock/models/stock_quant.py]: try to validate the domain

File: addons/stock/models/stock_quant.py [hunk 2]
```diff
@@ -173,6 +190,22 @@ def write(self, vals):
             return super(StockQuant, self).write(vals)
         return super(StockQuant, self).write(vals)
 
+    def fields_view_get(self, view_id=None, view_type='form', toolbar=False, submenu=False):
+        res = super().fields_view_get(view_id=view_id, view_type=view_type, toolbar=toolbar, submenu=submenu)
+        if res['type'] == 'search' and self.env.context.get('onhand_filter_domain'):
+            doc = etree.XML(res['arch'])
+            magic_filter_attrib = {
```
[the remark this answers is not in the document]
Comment (Whenrow) [reply addons/stock/models/stock_quant.py]: could you add a comment to explain what you are doing here ?

**Comment by sle-odoo** [comment]
@Whenrow can you double check the last commit? thanks :)

File: addons/stock/models/stock_quant.py [hunk 3]
```diff
@@ -130,15 +130,7 @@ def _set_inventory_quantity(self):
...
-        domain_loc = []
-        if self.env.context.get('active_model') == 'product.template':
-            product_tmpl = self.env['product.template'].browse(self.env.context.get('active_id'))
-            domain_loc = product_tmpl.product_variant_ids._get_domain_locations()[0]
-        elif self.env.context.get('active_model') == 'product.product':
-            product = self.env['product.product'].browse(self.env.context.get('active_id'))
-            domain_loc = product._get_domain_locations()[0]
-        else:
-            return domain_loc
+        domain_loc = self.env['product.product']._get_domain_locations()[0]
```
Comment (Whenrow) [inline addons/stock/models/stock_quant.py]: don't you, at least, filter on internal location ?

Comment (sle-odoo) [reply addons/stock/models/stock_quant.py]: that's the job of _get_domain_locations

**Comment by sle-odoo** [comment]
robodoo r+ rebase-ff

**Comment by sle-odoo** [comment]
robodoo retry

**Comment by sle-odoo** [comment]
robodoo retry

**Comment by sle-odoo** [comment]
robodoo retry

**Comment by sle-odoo** [comment]
robodoo retry

**Comment by Xavier-Do** [comment]
Yeayh!

**Pull request closed without merging**